src/EventBus/Extensions/EventBusBuilderExtensions.cs

```diff
@@ -25,34 +25,7 @@ public static IEventBusBuilder ConfigureJsonOptions(this IEventBusBuilder eventB
         // the consumer can use IKeyedServiceProvider.GetKeyedService<IIntegrationEventHandler>(typeof(T)) to get all
         // handlers for the event type.
 
-        // MediatR assembly scanning breaks with keyed services see https://github.com/jbogard/MediatR/issues/942.
-        // eventBusBuilder.Services.AddKeyedTransient<IIntegrationEventHandler, TH>(typeof(T));
-
-        // Instead, we'll register the handler as transient and use the HandlerTypes on SubscriptionInfo to get all
-        // handlers for the event type.
-        eventBusBuilder.Services.AddTransient<TH>();
-
-        eventBusBuilder.Services.Configure<EventBusSubscriptionInfo>(o =>
```

Comment:
We still need some of this, but only the list of event types.

This part:

o.EventTypes[typeof(T).Name] = typeof(T);

Reply:
Ah right. Just don't need the `HandlerTypes` business anymore.